### PR TITLE
feat: able to add extra envs to UI container

### PR DESCRIPTION
**Pull request by caiofralmeida**
Able to add extra envs to UI container configuration through values files.

related to the issue https://github.com/eshepelyuk/cmak-operator/issues/40

### CODE REVIEW

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
1. As soon as you change / add values - you should adjust values schema. Also it will be great if you can add corresponding linter tests, check `test/linter` folder.
2. Also, since you changed templates - please add helm unit tests , under `test/unit` folder.
3. Plz name new values as `extraEnv` - it is more wide spread in helm charts, so more common.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
please refer to contributing page regarding the format of commit messages.

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
The most important change is theformat of new values variable. It should not be just an object of values that are transformed into name / value pairs.
Instead int must be a **list** of objects, where each object is reflecting a one of the possible format of declaring pod env vars according to k8s doc, i.e. name/value, env from configmap, anv from secret etc.

File: templates/deployment.yaml [hunk 1]
```diff
@@ -77,10 +77,16 @@ spec:
           {{- range .Values.ui.extraArgs }}
           - {{ . | quote }}
           {{- end }}
-          {{ if .Values.ui.consumerProperties }}
+          {{- if or .Values.ui.consumerProperties .Values.ui.extraEnvs }}
```
Comment (eshepelyuk) [inline templates/deployment.yaml:80]: Plz move `env` outside of any if condition, i.e. make it render always.

**Review by eshepelyuk — COMMENTED** [review]
Also I would be very appreciated if you can modify the readme, by adding an alternative approach to configure basic auth via secret and env vars, 

https://github.com/eshepelyuk/cmak-operator#cmak-application-settings

**Comment by caiofralmeida** [comment]
@eshepelyuk thanks for the review and feedback, I'm going to fix it.

**Comment by eshepelyuk** [comment]
Hello @caiofralmeida 
What are yout interntions about this PR ?

**Comment by caiofralmeida** [comment]
@eshepelyuk Sorry for late, I'm out of work until 17th January. Maybe we can close this PR for now, and when I back, I can reopen with all requested changes.

**Comment by eshepelyuk** [comment]
Thanks for reply @caiofralmeida 
I just wanted to make sure that this PR is not abandoned.
So, as soon as you have time - let's proceed and finalize it.

**Comment by caiofralmeida** [comment]
Hi @eshepelyuk, is there a way to run unit tests locally?

**Review by eshepelyuk — CHANGES_REQUESTED** [review]
1. don't use existing unit test for a new test case, create a new file instead
1. the new test should have 2 cases
    - when extra env values are passed ( one you've provided)
    - when no extra env passed (default behaviour) 
1. your schema is  incorrect
    - default value must be empty list / array `[]`, not empty object `{}`
    - you should declare types of items in `extraEnv` array, in this case just use `object`
    - that also should be covered by linter tests

**Comment by eshepelyuk** [comment]
@caiofralmeida jFYI - I've updated `CONTRIBUTING` page, hopefully it became more useful.

**Comment by eshepelyuk** [comment]
Superseeded by #46

**Pull request closed without merging**